### PR TITLE
Add brain functions table for overview

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR is a follow-up of [#6460(comment)](https://github.com/mne-tools/mne-python/pull/6460#issuecomment-503474203) and adds a RST overview table from a functional point of view.
In my opinion, such a table improves visibility on the refactor/advancement of `mayavi`-based features of the `Brain` object.

I put all the functions I could find as a first step but it may be more useful to resize to the subset of functions used in practice and prioritize the tasks maybe as an ordered TODO list. Any feedback is welcome.

### CODE REVIEW

**Comment by agramfort** [comment]
no need to reimplement 100% of the API of pysurfer. See what breaks in the
doc
build with pyvista and you'll see what people have actually a chance to use.

>

**Comment by GuillaumeFavelier** [comment]
> no need to reimplement 100% of the API of pysurfer.

I agree with you @agramfort , we need only what's used of course.

> See what breaks in the doc build with pyvista and you'll see what people have actually a chance to use.

I like the idea and this was also suggested by @larsoner in this [comment](https://github.com/mne-tools/mne-python/pull/6460#issuecomment-503190977) I think.

But as the refactor is not yet completed, would you be okay to start with this version of the table and refine it during the process?

**Comment by agramfort** [comment]
sure as long as the objective is not to tick all boxes

>

**Comment by GuillaumeFavelier** [comment]
The objective is to tick the boxes we need.

File: mne/viz/_brain/_brain.py [hunk 1]
```diff
@@ -80,6 +80,109 @@ class _Brain(object):
...
+
+    Notes
+    -----
+    This table shows the capabilities of each Brain backend ("✓" for full
+    support, and "-" for partial support):
+
+    .. table::
+       :widths: auto
+
+    +--------------------------------+--------------+-----------------------+
```
Comment (larsoner) [inline mne/viz/_brain/_brain.py]: Doesn't this need to be indented three more spaces to be part of the above `table` directive?

**Pull request merged**